IPython/core/completer.py

```diff
@@ -2073,14 +2069,22 @@ def _complete(self, *, cursor_line, cursor_pos, line_buffer=None, text=None,
         return text, _matches, origins, completions
         
     def fwd_unicode_match(self, text:str) -> Tuple[str, list]:
+        if self._names is None:
+            self._names = []
+            for c in range(0,0x10FFFF + 1):
+                try:
+                    self._names.append(unicodedata.name(chr(c)))
+                except ValueError:
+                    pass
+
         # initial code based on latex_matches() method
         slashpos = text.rfind('\\')
         # if text starts with slash
         if slashpos > -1:
             s = text[slashpos+1:]
-            candidates = [x for x in names if x.startswith(s)]
+            candidates = [x for x in self._names if x.startswith(s)]
```

Comment:
If I'm not mistaken, we only use variables starting with underscore when that variable is not intended to be used (please someone correct me if needed). I don't think this is the case, so please consider changing this.

Reply:
It depends on context, when set as an attribute on an object is to mean that the attribute (or method) should be considered as private. It is meant for use only by the object author, and not by the object users (or at least at their own risk).

Reply:
We keep private variables with the underscore then? Is this stated in any PEP or best practices documment? I guess we could add some tool on it to the CI tests if they are not already there :) but thanks for the explanation

Reply:
Yes it is in [pep8 ](https://www.python.org/dev/peps/pep-0008/#descriptive-naming-styles): 

> _single_leading_underscore: weak "internal use" indicator. E.g. from M import * does not import objects whose names start with an underscore.

It is a convention. Nothing enforces it.

Reply:
Nice! I guess we could include some code style checks, something to discuss in a thread :)